### PR TITLE
[BXMSPROD-1892] Automated pull request backporting workflow

### PR DESCRIPTION
**Pull request by lampajr**
**JIRA**: 

https://issues.redhat.com/browse/BXMSPROD-1892

**referenced Pull Requests**:

* https://github.com/kiegroup/kogito-pipelines/pull/743

Added GitHub workflow to automate pull request backporting:
* create backporting pull requests whenever a pull request (having one or more labels `backport-<branch>`) is successfully merged.

### CODE REVIEW

**Review by radtriste — APPROVED** [review]

**Comment by mareknovotny** [comment]
looks good, Could you please add also a guidance into [PR template](https://github.com/kiegroup/drools/blob/main/.github/pull_request_template.md)?

**Comment by lampajr** [comment]
> looks good, Could you please add also a guidance into [PR template](https://github.com/kiegroup/drools/blob/main/.github/pull_request_template.md)?

@mareknovotny added guidance [here](https://github.com/kiegroup/drools/pull/4863/files#diff-b2496e80299b8c3150b1944450bd81c622e04e13d15c411d291db0927d75fd6bR100), do you agree? if so I will replicate it on all other repositories

**Review by Ginxo — CHANGES_REQUESTED** [review]
I would add a new section about "What's backporting" and create a link to it from  "How to backport a pull request to a different branch?" section. The rest LGTM

**Comment by lampajr** [comment]
> I would add a new section about "What's backporting" and create a link to it from "How to backport a pull request to a different branch?" section. The rest LGTM

@Ginxo Instead of adding a new section, wdyt if I directly add a simple definition of what a backporting is inside the `How to backport a pull request to a different branch` section? I think this way is cleaner and avoid to add too many stuff in the template itself.

Check this https://github.com/kiegroup/drools/pull/4863/files#diff-b2496e80299b8c3150b1944450bd81c622e04e13d15c411d291db0927d75fd6bR108

**Review by mareknovotny — APPROVED** [review]

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -97,3 +97,19 @@ How to retest this PR or trigger a specific build:
...
 </details>
+
+<details>
+<summary>
+How to backport a pull request to a different branch?
+</summary>
+
+In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).
+
+> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch.
```
Comment (Ginxo) [inline .github/pull_request_template.md]:
```suggestion
> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move an specific change or a set of them from one branch to another.
```

**Review by Ginxo — APPROVED** [review]
LGTM, just check my proposal

File: .github/pull_request_template.md (hunk 1 above)
Comment (lampajr) [reply .github/pull_request_template.md]: Agree with that, change applied

**Pull request merged**